src/game_map.cpp

```diff
@@ -1730,8 +1730,9 @@ void Game_Map::Parallax::ResetPositionX() {
 	}
 
 	if (!params.scroll_horz && !LoopHorizontal()) {
-		if (GetWidth() > 20 && parallax_width > Player::screen_width) {
-			const int w = (GetWidth() - 20) * TILE_SIZE;
+		int nbTilesPerScreen = Player::screen_width / TILE_SIZE;
```

Comment:
Thanks for tracking down what is going wrong. This and your ``14`` gave me a hint to work with.

To get the correct value this division must be float and rounded up.

Examples:

```
400 / 16 = 25 <- ok
399 / 16 = 24 <- bad, must be 25
385 / 16 = 24 <- bad, must be 25
384 / 16 = 24 <- ok
```

Though this is inefficient. Thankfully this is a power of two so much faster bit trickery to the rescue:

```
int tiles_per_screen = ((Player::screen_width & ~0xF) + 16) / TILE_SIZE;
```

This AND-masks the lowest 4 Bit away, so 399 becomes 384 and then add 16 to reach 400.